File: services/crud/src/main/java/com/blog/crud/blog/Blog.java
package com.blog.crud.blog;

import com.blog.crud.category.Category;
import com.blog.crud.comment.Comment;
import jakarta.persistence.*;
import lombok.*;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.time.LocalDateTime;
import java.util.List;

@Entity
@Getter
@Setter
@Builder
@AllArgsConstructor
@NoArgsConstructor
@EntityListeners(AuditingEntityListener.class)
public class Blog {
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "blog_seq")
    @SequenceGenerator(name = "blog_seq", sequenceName = "blog_sequence", allocationSize = 50)
    private Integer id;

    @Column(nullable = false, length = 100)
    private String title;

    private String author;

    @Column(nullable = false)
    private String content;

    @ManyToOne
    @JoinColumn(name = "category_id")
    private Category category;

    @OneToMany(fetch = FetchType.LAZY, mappedBy = "blog")
    private List<Comment> comments;

    @CreatedDate
    private LocalDateTime createdAt;

    @LastModifiedDate
    private LocalDateTime lastModifiedAt;
}

File: services/feed/models/blog.go
package models

import "time"

type Blog struct {
	ID             int        `db:"id" json:"id"`
	Title          string     `db:"title" json:"title"`
	Author         string     `db:"author" json:"author"`
	Content        string     `db:"content" json:"content"`
	IsPaid         bool       `db:"is_paid" json:"is_paid"`
	CategoryID     int        `db:"category_id" json:"category_id"`
	CoverImage     *string    `db:"cover_image" json:"cover_image"`
	CreatedAt      *time.Time `db:"created_at" json:"created_at"`
	LastModifiedAt *time.Time `db:"last_modified_at" json:"last_modified_at"`
}
